### PR TITLE
Add: stardust protostar

### PR DESCRIPTION
**Pull request by Owen-sz**
WIP

### CODE REVIEW

**Comment by Owen-sz** [comment]
I don't know why this is failing, could be due to the multiple results after it compiles..?

File: anda/stardust/non-spatial-input/stardust-non-spatial-input.spec [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Name:           stardust-non-spatial-input
+Version:        %commit_date.%shortcommit
+Release:        1%?dist
+Summary:        Tools you can easily snap together to get non-spatial input into stardust.
+URL:            https://github.com/StardustXR/non-spatial-input
+Source0:        %url/archive/%commit/non-spatial-input-%commit.tar.gz
+License:        MIT
+BuildRequires:  cargo cmake anda-srpm-macros cargo-rpm-macros mold libudev-devel g++ libinput-devel libxkbcommon-x11-devel
+Requires:       libgcc glibc
```
Comment (madonuko) [inline anda/stardust/non-spatial-input/stardust-non-spatial-input.spec]: don't need this

File: anda/stardust/non-spatial-input/stardust-non-spatial-input.spec [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+%prep
+%autosetup -n non-spatial-input-%commit
+%cargo_prep_online
+
+%build
+%cargo_build
+
+%install
+%cargo_install
```
Comment (madonuko) [inline anda/stardust/non-spatial-input/stardust-non-spatial-input.spec]: here you are building all the projects in that repository

either `cd` into each of them to run `%cargo_install` (+ remove %cargo_build), or manually run the `install -Dpm755` command

File: anda/stardust/non-spatial-input/update.rhai [hunk 3]
```diff
@@ -0,0 +1,7 @@
+if filters.contains("nightly") {
```
Comment (madonuko) [inline anda/stardust/non-spatial-input/update.rhai]: don't need this

File: anda/stardust/non-spatial-input/stardust-non-spatial-input.spec [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+%files
+%_bindir/non-spatial-input
```
Comment (madonuko) [inline anda/stardust/non-spatial-input/stardust-non-spatial-input.spec]: no such project/file

File: anda/stardust/non-spatial-input/stardust-non-spatial-input.spec [hunk 5]
```diff
@@ -0,0 +1,40 @@
+%global commit 5ac7f04f6876097aa8c3cf9af033d609a8a49944
+%global commit_date 20240824
+%global shortcommit %(c=%{commit}; echo ${c:0:7})
+%global debug_package %{nil}
```
Comment (madonuko) [inline anda/stardust/non-spatial-input/stardust-non-spatial-input.spec]: bad idea for a rust package, remove this line

File: anda/stardust/protostar/update.rhai [hunk 6]
```diff
@@ -0,0 +1,7 @@
+if filters.contains("nightly") {
```
Comment (madonuko) [inline anda/stardust/protostar/update.rhai]: again don't need this

File: anda/stardust/protostar/stardust-protostar.spec [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+Tools you can easily snap together to get non-spatial input into stardust.
+
+%prep
+%autosetup -n protostar-%commit
+%cargo_prep_online
+
+%build
+
+%install
+%cargo_install
```
Comment (madonuko) [inline anda/stardust/protostar/stardust-protostar.spec]:
```suggestion
cd protostar
%cargo_install
```
probably this? I assume that's what you want to do?

File: anda/stardust/protostar/stardust-protostar.spec [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+Name:           stardust-protostar
+Version:        %commit_date.%shortcommit
+Release:        1%?dist
+Summary:        Tools you can easily snap together to get non-spatial input into stardust.
+URL:            https://github.com/StardustXR/protostar
+Source0:        %url/archive/%commit/protostar-%commit.tar.gz
+License:        MIT
+BuildRequires:  cargo cmake anda-srpm-macros cargo-rpm-macros mold libudev-devel g++ libinput-devel libxkbcommon-x11-devel
+Requires:       libgcc glibc
```
Comment (madonuko) [inline anda/stardust/protostar/stardust-protostar.spec]: don't need this

**Review by madonuko — CHANGES_REQUESTED** [review]

**Review by madonuko — DISMISSED** [review]

**Review by madonuko — APPROVED** [review]

**Comment by raboneko** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|f39|[](https://github.com/terrapkg/packages/pull/2075)|
|✅|f40|[](https://github.com/terrapkg/packages/pull/2076)|
|✅|f41|[](https://github.com/terrapkg/packages/pull/2077)|
|✅|el9|[](https://github.com/terrapkg/packages/pull/2078)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport) and see the [Github Action logs](https://github.com/terrapkg/packages/actions/runs/10890643048) for details

**Pull request merged**